intelmq/bin/intelmqsetup.py

```diff
@@ -17,59 +22,135 @@
+FILE_OUTPUT_PATH = os.path.join(VAR_STATE_PATH, 'file-output/')
 
+
+def basic_checks(skip_ownership):
+    if os.geteuid() != 0 and not skip_ownership:
+        sys.exit(red('You need to run this program as root for setting file ownership!'))
     if not ROOT_DIR:
-        sys.exit('Not a pip-installation of IntelMQ, nothing to initialize.')
-
-    create_dirs = ('%s/file-output' % VAR_STATE_PATH,
-                   VAR_RUN_PATH,
-                   DEFAULT_LOGGING_PATH,
-                   CONFIG_DIR)
-    for create_dir in create_dirs:
-        if not os.path.isdir(create_dir):
-            os.makedirs(create_dir, mode=0o755,
-                        exist_ok=True)
-            print('Created directory %r.' % create_dir)
+        sys.exit(red('Not a pip-installation of IntelMQ, nothing to initialize.'))
+
+    if skip_ownership:
+        return
+    try:
+        getpwnam('intelmq')
+    except KeyError:
+        sys.exit(red("User 'intelmq' does not exist. Please create it and then re-run this program."))
+    try:
+        getgrnam('intelmq')
+    except KeyError:
+        sys.exit(red("Group 'intelmq' does not exist. Please create it and then re-run this program."))
+
+
+def create_directory(directory: str, octal_mode: int, readable_mode: str):
+    if not os.path.isdir(directory):
+        os.makedirs(directory, mode=octal_mode,
+                    exist_ok=True)
+        print(f'Created directory {directory!r} with permissions {readable_mode}.')
+    else:
+        actual_mode = stat.filemode(os.stat(directory).st_mode)
+        if actual_mode != readable_mode:
+            print(f'Fixed wrong permissions of {directory!r}: {actual_mode!r} -> {readable_mode!r}.')
+            os.chmod(directory, octal_mode)
+
+
+def change_owner(file: str, owner=None, group=None):
+    if owner and getpwuid(os.stat(file).st_uid).pw_name != owner:
+        print(f'Fixing owner of directory {file!r}.')
+        shutil.chown(file, user=owner)
+    if group and getpwuid(os.stat(file).st_gid).pw_name != group:
+        print(f'Fixing group of directory {file!r}.')
+        shutil.chown(file, group=group)
+
+
+def find_webserver_user():
+    candidates = ('www-data', 'wwwrun', 'httpd', 'apache')
+    for candidate in candidates:
+        try:
+            getpwnam(candidate)
+        except KeyError:
+            pass
+        else:
+            print(f'Detected webserver username {candidate!r}.')
+            return candidate
+
+
+def intelmqsetup_core(ownership=True, state_file=STATE_FILE_PATH):
+    directories_modes = ((FILE_OUTPUT_PATH, 0o755, 'drwxr-xr-x'),
+                         (VAR_RUN_PATH, 0o755, 'drwxr-xr-x'),
+                         (DEFAULT_LOGGING_PATH, 0o755, 'drwxr-xr-x'),
+                         (CONFIG_DIR, 0o775, 'drwxrwxr-x'),
+                         )
+    for directory, octal_mode, readable_mode in directories_modes:
+        create_directory(directory, octal_mode, readable_mode)
```

Comment:
There are 4 directories- so if you simply write `create_directory` for every directory you have 4 lines of code and it is readable. Why put the directories in a list of tuples and then iterate through them? IMO that makes it only more complex.